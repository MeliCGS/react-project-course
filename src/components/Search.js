import { useState } from 'react'
import  Datos  from './Datos'
import ListMovie from './ListMovie'


function Search(props) {

const [ search, setSearch] = useState('')
const [ results, setResults ] = useState([])

const _handleChange = (e) => {
  setSearch(e.target.value)
}

const _handleResults = (results) => {
    setResults({results})
  }

const _handleSubmit = () => {
    Datos(search, _handleResults)
}

    return (
        <div>
            <input role="input" onChange={_handleChange} placeholder="escriba aquí" type="text"/>
            <button role="button" onClick={_handleSubmit}>Buscar</button>
            {
                results.length === 0
                ? <h3>Introduzca un título</h3>
                : <ListMovie movies={results} />
            }
        </div>
    )

}

export default Search