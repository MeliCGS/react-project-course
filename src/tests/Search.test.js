import React from 'react'
import '@testing-library/jest-dom/extend-expect'
import Search from '../components/Search'
import { screen } from '@testing-library/dom'
import { render } from '@testing-library/react'

describe('Pruebas en el componente Search', () => {
   beforeEach(() => {
      render(<Search />)
    });

   test('El componente Search se renderiza', () => {
      const { container } = render(<Search />)
      expect(container).toMatchSnapshot()
   })
   test('El componente Search deberia renderizar el Título', () => {
      const titleSearch = screen.getByText(/Introduzca un título/i)
      expect(titleSearch).not.toBeFalsy()
   })
   test('El componente Search deberia incluir el texto del boton', () => {
      const btnSearch = screen.getByText(/Buscar/i)
      expect(btnSearch).not.toBeFalsy()
   })
   test('El componente Search deberia incluir un elemento input', () => {
      const getInput = screen.getByRole("input")
      expect(getInput).toBeInTheDocument()
   })
   test('En el input del Search deberia contener un texto placeHolder', () => {
      const text = "escriba aquí"
      const getInput = screen.getByPlaceholderText('escriba aquí')
      expect(getInput).toBeInTheDocument()
   })
})

