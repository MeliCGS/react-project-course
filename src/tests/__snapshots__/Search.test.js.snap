// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente Search El componente Search se renderiza 1`] = `
<div>
  <div>
    <input
      placeholder="escriba aquí"
      role="input"
      type="text"
    />
    <button
      role="button"
    >
      Buscar
    </button>
    <h1>
      Introduzca un título
    </h1>
  </div>
</div>
`;
